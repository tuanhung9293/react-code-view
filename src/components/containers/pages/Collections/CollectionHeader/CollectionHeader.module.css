.header {
    padding: 15px 0;
}

.title {
    font-size: 33px;
}

.addButton {
    text-align: right;
}