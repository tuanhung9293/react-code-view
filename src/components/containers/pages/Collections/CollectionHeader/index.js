import React from 'react'
import {connect} from 'react-redux'
import {objectToUrlParams} from 'services//transformers'
import {Columns, Column} from 'components/ui/grid'
import PrimaryButton from 'components/ui/buttons/PrimaryButton'
import styles from './CollectionHeader.module.css'

const CollectionHeader = ({searchParams}) => {
  return (
    <div className={styles.header}>
      <Columns>
        <Column span={8} breakPoints={[{maxWidth: 576, span: 12}]}>
          <div className={styles.title}>NASA COLLECTION</div>
        </Column>
        <Column span={4} breakPoints={[{maxWidth: 576, span: 12}]}>
          <div className={styles.addButton}>
            <PrimaryButton to={`/nasa-search${objectToUrlParams(searchParams)}`}>
              Add new item
            </PrimaryButton>
          </div>
        </Column>
      </Columns>
    </div>
  )
}

export default connect(
  ({medias}) => ({searchParams: medias.params})
)(CollectionHeader)
