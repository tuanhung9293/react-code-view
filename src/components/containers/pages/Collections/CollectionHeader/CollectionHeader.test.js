import React from 'react'
import {mount, shallow} from 'enzyme'
import {MemoryRouter} from 'react-router-dom'
import {Provider} from 'react-redux'
import CollectionHeader from './'

import configureStore from 'redux-mock-store'
const mockStore = configureStore()

let store = mockStore({
  medias: {
    params: {
      q: 'data',
      mediaType: 'audio'
    }
  },
  ui: {windowSize: {width: 615, height: 953}}
})
store.dispatch = () => {}


describe('renders', () => {
  it('renders CollectionHeader', () => {
    expect(shallow(
      <Provider store={store}>
        <CollectionHeader />
      </Provider>
    )).toMatchSnapshot()
  })

  it('renders CollectionHeader Component', () => {
    jest.mock('../../../../ui/buttons/PrimaryButton', () => ()=> <div id="mockPrimaryButton">
      PrimaryButton
    </div>)

    const wrapper = mount(
      <Provider store={store}>
        <MemoryRouter initialEntries={[{pathname: '/', key: 'testKey'} ]}>
          <CollectionHeader />
        </MemoryRouter>
      </Provider>
    )
    expect(wrapper.find(CollectionHeader)).toMatchSnapshot()
  })
})

