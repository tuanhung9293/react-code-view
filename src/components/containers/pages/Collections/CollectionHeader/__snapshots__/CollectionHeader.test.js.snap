// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders renders CollectionHeader 1`] = `<Connect(CollectionHeader) />`;

exports[`renders renders CollectionHeader Component 1`] = `
<Connect(CollectionHeader)>
  <CollectionHeader
    dispatch={[Function]}
    searchParams={
      Object {
        "mediaType": "audio",
        "q": "data",
      }
    }
  >
    <div
      className="header"
    >
      <Columns>
        <div
          className="base"
        >
          <Connect(Column)
            breakPoints={
              Array [
                Object {
                  "maxWidth": 576,
                  "span": 12,
                },
              ]
            }
            span={8}
          >
            <Column
              breakPoints={
                Array [
                  Object {
                    "maxWidth": 576,
                    "span": 12,
                  },
                ]
              }
              dispatch={[Function]}
              span={8}
              windowWidth={615}
            >
              <div
                className="base grid"
                style={
                  Object {
                    "WebkitBoxFlex": 0,
                    "flexBasis": "66.66666666666667%",
                    "flexGrow": 0,
                    "maxWidth": "66.66666666666667%",
                    "msFlexPositive": 0,
                    "msFlexPreferredSize": "66.66666666666667%",
                  }
                }
              >
                <div
                  className="title"
                >
                  NASA COLLECTION
                </div>
              </div>
            </Column>
          </Connect(Column)>
          <Connect(Column)
            breakPoints={
              Array [
                Object {
                  "maxWidth": 576,
                  "span": 12,
                },
              ]
            }
            span={4}
          >
            <Column
              breakPoints={
                Array [
                  Object {
                    "maxWidth": 576,
                    "span": 12,
                  },
                ]
              }
              dispatch={[Function]}
              span={4}
              windowWidth={615}
            >
              <div
                className="base grid"
                style={
                  Object {
                    "WebkitBoxFlex": 0,
                    "flexBasis": "33.333333333333336%",
                    "flexGrow": 0,
                    "maxWidth": "33.333333333333336%",
                    "msFlexPositive": 0,
                    "msFlexPreferredSize": "33.333333333333336%",
                  }
                }
              >
                <div
                  className="addButton"
                >
                  <PrimaryButton
                    to="/nasa-search?q=data&mediaType=audio"
                  >
                    <Button
                      className="base"
                      to="/nasa-search?q=data&mediaType=audio"
                    >
                      <Link
                        className="base"
                        replace={false}
                        to="/nasa-search?q=data&mediaType=audio"
                      >
                        <a
                          className="base"
                          href="/nasa-search?q=data&mediaType=audio"
                          onClick={[Function]}
                        >
                          Add new item
                        </a>
                      </Link>
                    </Button>
                  </PrimaryButton>
                </div>
              </div>
            </Column>
          </Connect(Column)>
        </div>
      </Columns>
    </div>
  </CollectionHeader>
</Connect(CollectionHeader)>
`;
